pmix RC's fail to build on ArmV8 (ThunderX)

## Background information
All current packages including RC's current fail to build on thunderx (arm 64 bit)

The current git 9b340496c6   does build,  this is more of a notice of bugs with the current rc's and releases.

```
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:304:(.text+0xa44): relocation truncated to fit: R_AARCH64_LDST32_ABS_LO12_NC against symbol `spawn_wait' defined in .bss section in pmix_test.o
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:304: warning: One possible cause of this error is that the symbol is being referenced in the indicated code as if it had a larger alignment than was declared where it was defined.
server_callbacks.o: In function `release_cb':
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:287:(.text+0xab8): relocation truncated to fit: R_AARCH64_LDST32_ABS_LO12_NC against symbol `spawn_wait' defined in .bss section in pmix_test.o
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:287: warning: One possible cause of this error is that the symbol is being referenced in the indicated code as if it had a larger alignment than was declared where it was defined.
server_callbacks.o: In function `_release_cb':
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:278:(.text+0xb00): relocation truncated to fit: R_AARCH64_LDST32_ABS_LO12_NC against symbol `spawn_wait' defined in .bss section in pmix_test.o
/root/rpmbuild/BUILD/pmix-2.1.1rc1/test/server_callbacks.c:278: warning: One possible cause of this error is that the symbol is being referenced in the indicated code as if it had a larger alignment than was declared where it was defined.
```

### What version of the PMIx Reference Library are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)

2.0.2
2.1.1rc1
2.0.3rc3


### Describe how PMIx was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

`rpmbuild -ta pmix-2.1.1rc1.tar.bz2  -D 'install_in_opt 1'`
`./configure && make -j 96 `

### Please describe the system on which you are running

* Operating system/version: RHEL 7.3 and CentOS 7.2 aarch64
* Computer hardware:  Cavium ThunderX  aarch64
* Network type:  40Gig Ethernet

-----------------------------

## Details of the problem

Please describe, in detail, the problem that you are having, including the behavior you expect to see, the actual behavior that you are seeing, steps to reproduce the problem, etc.  It is most helpful if you can attach a small program that a developer can use to reproduce your problem.

**Note**: If you include verbatim output (or a code block), please use a [GitHub Markdown](https://help.github.com/articles/creating-and-highlighting-code-blocks/) code block like below:
```shell
shell$ prun -np 2 ./hello_world
```

--- a/test/server_callbacks.c
+++ b/test/server_callbacks.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015-2017 Intel, Inc. All rights reserved.
+ * Copyright (c) 2015-2018 Intel, Inc. All rights reserved.
  * Copyright (c) 2015      Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
  * Copyright (c) 2015      Mellanox Technologies, Inc.
@@ -18,7 +18,7 @@
 #include "server_callbacks.h"
 #include "src/util/argv.h"
 
-extern int spawn_wait;
+extern bool spawn_wait;
 
 pmix_server_module_t mymodule = {
     .client_connected = connected,

